social-tw/social-tw-website
# feat: add logout function

## Summary

In this Pull Request (PR), I have implemented a logout functionality for our application. This functionality enables a user to successfully log out from the application, clear user data, and return to the initial state.

## Linked Issue

There isn't a specific issue linked to this PR.

## Details

The implementation involves using React's Context API to manage the user state. A logout method is provided in the User class within the User context, which sets the `hasSignedUp` attribute to false, clears the user's state and removes the user's credentials from the local storage. 

## Impacted Areas

The files impacted by this PR are:
- `UserDropdown.tsx`: In this file, I added a click handler to the Logout button which triggers the logout function from the UserContext.
- `User.ts`: The logout method was added to the User class in this file.

## Tests

I have tested the logout functionality manually to ensure it works as expected. No specific environment or configuration is needed to run these tests.

## Possible Impacts

The changes made in this PR should not have any backward compatibility issues or affect any existing features negatively. However, all features that rely on user state should be thoroughly tested to ensure they handle a logged out state properly.

## Visual Materials

The PR does not involve any significant UI changes and hence no screenshots or videos are attached.

## Verification Steps

To verify the changes, follow these steps:
1. Sign in to the application.
2. Click on the User icon to reveal the dropdown.
3. Click on the 'Logout' option.
4. The user should now be logged out, the user state cleared, and user credentials removed from local storage.

## Todo

This PR is complete and doesn't require any additional steps.
- [x] Implement logout functionality in `UserDropdown.tsx` and `User.ts`
- [x] Test logout functionality manually

## Checklist
